app/routes.py

```diff
@@ -1,2 +1,97 @@
-from flask import Blueprint
+from app import db
+from app.models.planet import Planet
+from flask import Blueprint, jsonify, make_response, request, abort
+
+
+planet_bp = Blueprint("planet", __name__, url_prefix="/planets")
+
+
+@planet_bp.route("", methods=["GET"])
+def read_all_planets():
+    habitable_query = request.args.get("habitable")
+    if habitable_query:
+        planets = Planet.query.filter_by(habitable = habitable_query)
+
+    else:
+        planets = Planet.query.all()
+    planets_response = []
+
+    for planet in planets:
+        planets_response.append({
+            "id" : planet.id,
+            "name" : planet.name,
+            "description" : planet.description,
+            "habitable" : planet.habitable
+        })
+    return jsonify(planets_response)
+
+
+@planet_bp.route("", methods=["POST"])
+def create_one_planet():
+    request_body = request.get_json()
+
+    new_planet = Planet(name = request_body["name"],
+    description = request_body["description"],
+    habitable = request_body["habitable"])
+
+    db.session.add(new_planet)
+    db.session.commit()
+
+    return make_response(f"Planet {new_planet.name} successfully created!", 201)
+
+
+@planet_bp.route("/<id>", methods=["GET"])
+def read_one_planet(id):
+
+    planet = validate_planet(id)
+
+
+    return {
+            "id" : planet.id,
+            "name" : planet.name,
+            "description" : planet.description,
+            "habitable" : planet.habitable
+        }
+
+
+@planet_bp.route("/<id>", methods=["PUT"])
+def update_one_planet(id):
+
+    planet = validate_planet(id)
+    request_body = request.get_json()
+
+    planet.name = request_body["name"]
+    planet.description = request_body["description"]
+    planet.habitable = request_body["habitable"]
+
+    db.session.commit()
+
+    return make_response(f"Planet {planet.id} successfully updated.")
```

Comment:
while `make_response` will send a default code back with the response, it doesn't hurt to get in the habit of controlling our responses in a predictable way. If we sent a response code on line 40, then we should keep doing that will all our routes